google-http-client-xml/src/test/java/com/google/api/client/xml/AtomTest.java

```diff
@@ -44,8 +75,205 @@ public void subtestSetSlugHeader(HttpHeaders headers, String expectedValue, Stri
+    XmlNamespaceDictionary namespaceDictionary = new XmlNamespaceDictionary();
+    Xml.parseElement(parser, feed, namespaceDictionary, null);
+    assertNotNull(feed);
+    assertEquals(2, feed.entry.length);
+
+    assertEquals("John Doe", feed.author.name);
+    assertEquals("urn:uuid:60a76c80-d399-11d9-b93C-0003939e0af6", feed.id);
+    assertEquals("2003-12-13T18:31:02Z", feed.updated);
+    assertEquals("Example Feed", feed.title);
+    assertEquals("http://example.org/", feed.link.href);
+
+    FeedEntry entry1 = feed.entry[0];
+    //assertNotNull(feed.entry);
+    assertEquals("urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa6a", entry1.id);
+    assertEquals("2003-12-13T18:30:02Z", entry1.updated);
+    assertEquals("Some text.", entry1.summary);
+    assertEquals("Atom-Powered Robots Run Amok", entry1.title);
+    assertEquals("http://example.org/2003/12/13/atom03", entry1.link.href);
+
+    FeedEntry entry2 = feed.entry[1];
+    assertEquals("urn:uuid:1225c695-cfb8-4ebb-aaaa-80da344efa62", entry2.id);
+    assertEquals("2003-12-13T18:32:02Z", entry2.updated);
+    assertEquals("Some other text.", entry2.summary);
+    assertEquals("Atom-Powered Robots Run Amok!", entry2.title);
+    assertEquals("http://example.org/2003/12/13/atom02", entry2.link.href);
+  }
+
+  /**
+   * Reading an XML ATOM Feed from a file and valid if all the {@link FeedEntry} are present. No
+   * detailed evaluation of each element
+   *
+   * The purpose of this test is to read a bunch of elements which contain additional elements
+   * (HTML in this case), that are not part of the {@link FeedEntry} and to see if there is an issue
+   * if we parse some more entries.
+   */
+  @SuppressWarnings("unchecked")
+  @Test
+  public void testHeiseFeedParser() throws Exception {
+    XmlPullParser parser = Xml.createParser();
+    String read = readFile("heise-atom.xml");
+    parser.setInput(new StringReader(read));
+    XmlNamespaceDictionary namespaceDictionary = new XmlNamespaceDictionary();
+    AbstractAtomFeedParser atomParser = new AtomFeedParser<Feed, FeedEntry>(namespaceDictionary,
+        parser, new ByteArrayInputStream(read.getBytes()), Feed.class, FeedEntry.class);
+    Feed feed = (Feed) atomParser.parseFeed();
+    assertNotNull(feed);
+
+    int counter = 0;
+    while (atomParser.parseNextEntry() != null) {
+      counter++;
+    }
+
+    atomParser.close();
+    assertEquals(62, counter);
+  }
+
+  /**
+   * We need to method to read the file into a string
+   *
+   * @param file file name in the resource folder that will be parsed
+   * @return content of the File as String.
+   * @throws IOException in case the file was not able to be parsed
+   * @see <a href="https://goo.gl/CJ4v7Z">Stackoverflow</a>
+   */
+  private String readFile(String file) throws IOException {
+    ClassLoader classLoader = getClass().getClassLoader();
+    BufferedReader reader = new BufferedReader(new FileReader(classLoader.getResource(file)
```

Comment:
This needs an explicit encoding. Don't rely on the platform default encoding.

Reply:
Switched to guava implementation